src/routes/webhooks/paddle.ts

```diff
@@ -0,0 +1,105 @@
+import { FastifyInstance } from 'fastify';
+import {
+  Environment,
+  EventName,
+  Paddle,
+  SubscriptionCanceledEvent,
+  SubscriptionCreatedEvent,
+} from '@paddle/paddle-node-sdk';
+import createOrGetConnection from '../../db';
+import { updateSubscriptionFlags } from '../../common';
+import { User } from '../../entity';
+import { logger } from '../../logger';
+
+const paddleInstance = new Paddle(process.env.PADDLE_API_KEY, {
+  environment: Environment.sandbox,
+});
+
+const planTypes = {
+  pri_01jcdp5ef4yhv00p43hr2knrdg: 'monthly',
+  pri_01jcdn6enr5ap3ekkddc6fv6tq: 'yearly',
+};
+
+const updateUserSubscription = async ({
+  data,
+  state,
+}: {
+  data: SubscriptionCreatedEvent | SubscriptionCanceledEvent | undefined;
+  state: boolean;
+}) => {
+  if (!data) {
+    return;
+  }
+  const con = await createOrGetConnection();
+  const userId = data.data?.customData?.user_id;
+  if (!userId) {
+    logger.error('User ID missing in payload');
+    return false;
+  }
+  const subscriptionType = data.data?.items.reduce((acc, item) => {
+    if (planTypes[item.price?.id]) {
+      acc = planTypes[item.price?.id];
+    }
+    return acc;
+  }, null);
+  if (!subscriptionType) {
+    logger.error('Subscription type missing in payload');
+    return false;
+  }
+  await con.getRepository(User).update(
+    {
+      id: data.data?.customData?.user_id,
+    },
+    {
+      subscriptionFlags: updateSubscriptionFlags({
+        [subscriptionType]: state,
+      }),
```

Comment:
Logging to CIO later?

Reply:
Yeah still coming, need to check how I'm going to do it.